### PR TITLE
Coerce occ config:app:set boolish value to 0 or 1

### LINKED ISSUE #37697: config:app:set has no "type" parameter
`occ config:system:set` has `--type` parameter and you can set a setting to boolean true or false.

`occ config:app:set` has no `--type` parameter. Do something like:

```
php occ config:app:set files enable_lock_file_action --value true
```

Actually the setting is string "true". Code that gets the setting ends up understanding that it is boolean `true` because it is non-zero.

```
php occ config:app:set files enable_lock_file_action --value false
```

The setting is string "false". Code that gets the setting ends up understanding that it is boolean `true` because it is non-zero. But users get confused, it looks false but the functionality is still enabled.

```
php occ config:app:set files enable_lock_file_action --value 0
```
Setting to `0` really switched off the setting.

```
occ config:app:delete files enable_lock_file_action
```
Deleting the setting allows it to go back to its default of `false`.

It would be less confusing if we could do:
```
php occ config:app:set files enable_lock_file_action --type boolean --value false
```

If the backend database field is only storing a string, then `--type boolean --value false` could store "0" and `--type boolean --value true` could store "1".

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
Setting an app value via the occ config:app:set command to false may not result to a negation evaluation due the value will be stored in the database as 'false' type string. 
As this is a non-zero value it may evaluate to true.
To come over this problem we will coerce the value: 
true => 1
false => 0

## Related Issue

- Fixes <https://github.com/owncloud/core/issues/37697>

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [x] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]
unit test? THX

**Comment by HieronymusPerro** [comment]
> unit test? THX

Added a acceptance test instead

**Comment by jvillafanez** [comment]
I don't think this is a good idea.

There are a couple of things to take into account:
* The DB will store any value as string because that's how the DB column is defined.
* The user will always input a string

I guess the main problem here is the lack of information. This cli command is part of core and it very agnostic about how the value will be used.
The key point is that it depends on the app how that value will be interpreted, and as such, it should be documented by the app using the value.

For example, an app could store "useFeature" with "true" or "false", so the app can tell you to use `occ config:app:set app useFeature --value true`. Now, the app could have something like:
```
$useFeature = getFromDB('useFeature');
if ($useFeature === 'true') {
.....
}
```
so yeah, the app is explicitly checking for the "true" string because it knows that the function will return a string, and it expect it to be "true" the same it could expect it to be "theAppMustUseThis" string.
The app could also use the `filter_var` function for a more flexible validation so "1" or "on" could also be accepted as valid, but again, it depends on the app.

My main point is that the interpretation of the value depend on each specific app using the value, and core shouldn't interfere.

Note that this is different than the case with `occ config:system:set` where it will be translated to actual php code, which requires explicit typing.

**Comment by jvillafanez** [comment]
I agree that having "true" / "false" stored and used as strings is confusing and should be avoided, but it's probably a better idea to promote the usage of "yes" / "no" instead and check the value with `filter_var`.

**Comment by HieronymusPerro** [comment]
I mostly agree @micbar must decide

**Comment by mmattel** [comment]
@JanAckermann any news on that?

**Comment by JanAckermann** [comment]
Not yet @mmattel, I need to speak about this with @micbar 

More thoughts on this are welcome as well

File: changelog/unreleased/37697 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Enhancement: Fix expiring a wrong share entry problem
```
Comment (karakayasemi) [inline changelog/unreleased/37697:1]: This looks like a copy-paste mistake.

**Comment by JanAckermann** [comment]
@JammingBen time to say good bye to our real first PR :cry:

**Pull request closed without merging**